routes/eth.go

```diff
@@ -367,14 +369,104 @@ func (fes *APIServer) QueryETHRPC(ww http.ResponseWriter, req *http.Request) {
 	}
 }
 
+type MetamaskSignInRequest struct {
+	RecipientPublicKey  string
+	RecipientEthAddress string
```

Comment:
I think we should get rid of this parameter and instead compute it from `RecipientPublicKey`. Otherwise someone could just get infinite money by spamming this endpoint with random ETH addresses

Reply:
Hmm couldn't i also derive the RecipientPublicKey from the message and signature?